File: src/main/java/ru/sicampus/bootcamp2026/config/JwtConfig.java
package ru.sicampus.bootcamp2026.config;

import org.springframework.core.io.Resource;
import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Getter
@Setter
@Configuration
@ConfigurationProperties(prefix = "jwt")
public class JwtConfig {
    private Resource publicKey;
    private Resource privateKey;
    private long accessTokenExpirationMs;
    private long refreshTokenExpirationMs;
}

File: server/src/main/kotlin/com/nikshet/industrialbackend/config/JwtConfig.kt
package com.nikshet.industrialbackend.config

import com.nikshet.industrialbackend.providers.JwtTokensProvider
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class JwtConfig {

    @Bean
    fun tokensProvider(
        @Value($$"${jwt.secret}") secret: String,
        @Value($$"${jwt.access-token.expiration-ms:604800000}") accessTokenExpirationMs: Long,
        @Value($$"${jwt.refresh-token.expiration-ms:2592000000}") refreshTokenExpirationMs: Long
    ): JwtTokensProvider {
        return JwtTokensProvider(
            secret = secret,
            accessTokenExpirationMs = accessTokenExpirationMs,
            refreshTokenExpirationMs = refreshTokenExpirationMs
        )
    }
}